c/kastore.c

```diff
@@ -362,9 +362,17 @@ kastore_write_data(kastore_t *self)
             goto out;
         }
         size = self->items[j].array_len * type_size(self->items[j].type);
-        if (size > 0 && fwrite(self->items[j].array, size, 1, self->file) != 1) {
-            ret = KAS_ERR_IO;
-            goto out;
+        if (self->flags & KAS_PUT_BORROWS_ARRAY) {
```

Comment:
Can't we do something like this?
```c
const void *write_array;

item = self->items[j];
assert((item->borrowed_array == NULL) != (item->array == NULL));
write_array = item->borrowed_array != NULL? item->borrowed_array: item->array;
if (size > 0 && fwrite(write_array, size, 1, self->file) != 1) {
...
```
We shouldn't need to duplicate logic like this.